### PR TITLE
ufal/shibboleth-encode-redirecturl

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  | 2.5  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

File: src/aai/aai.js [hunk 1]
```diff
@@ -21,7 +21,24 @@
       var targetUrl = '';
       var opts = jQuery.extend({}, this.defaults, options),
           defaultCallback = function(e) {
-            targetUrl = opts.target + '?redirectUrl=' + window.location.href;
+            targetUrl = opts.target + '?redirectUrl=';
+            var redirectUrl = '';
+
+            // Redirection could be initiated from the login page; in that case,
+            // we need to retrieve the redirect URL from the URL parameters
+            var urlParams = new URLSearchParams(window.location.href.split('?')[1]);
```
Comment (vidiecan) [inline src/aai/aai.js]: can there be multiple `?`?

File: src/aai/aai.js [hunk 2]
```diff
@@ -21,7 +21,24 @@
...
+            var redirectUrlFromLogin = urlParams.get('redirectUrl') || null;
+
+            if (redirectUrlFromLogin == null || redirectUrlFromLogin === '') {
```
Comment (vidiecan) [inline src/aai/aai.js]: var redirectUrl = window.location.href;
if (XXX)
{
   append...
}
would be more readable?

File: src/aai/aai.js [hunk 3]
```diff
@@ -97,6 +114,24 @@
         return djc;
       } //if jQuery(selector)
     };
+
+    function getCookie(cookieName) {
```
Comment (vidiecan) [inline src/aai/aai.js]: isn't there a utility function that handles cookies? I have seen handling cookies multiple times

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: src/aai/aai.js (hunk 1 above)
Comment (milanmajchrak) [reply src/aai/aai.js]: Yes, https://stackoverflow.com/questions/2924160/is-it-valid-to-have-more-than-one-question-mark-in-a-url

File: src/aai/aai.js (hunk 2 above)
Comment (milanmajchrak) [reply src/aai/aai.js]: Refactored

File: src/aai/aai.js (hunk 3 above)
Comment (milanmajchrak) [reply src/aai/aai.js]: Removed. I forgot to remove it.

**Review by vidiecan — APPROVED** [review]

**Pull request merged**